labs/lab17-persistent-storage/readme.md

```diff
@@ -0,0 +1,815 @@
+
+It is split into 5 sections:
+
+- [The Example Application](#the-example-application)
+- [User-Facing Configuration](#user-facing-configuration)
+- [Embedding a Database](#embedding-a-database)
+- [Connecting to an External Database](#connecting-to-an-external-database)
+
+### Prerequisites
+
+We will be creating a few releases of our application. Creating releases is covered in the [Lab00-hello-world]((../lab00-hello-world)) lab, so make sure to complete it before starting this one.
+
+## The Example Application
+
+For demonstration purposes, we'll use a simple flask app that connects to a postgres database and retrieves the list of databases and displays them in a browser. 
+Once you've finished this guide, you should feel confident replacing it with any Kubernetes workload(s) that need to connect to a database.
+
+For this lab, the first release of this application has already been created and deployed to the VM that was provisioned for you. The files used to create the first release are also available in the [manifests](./manifests) folder. For reference, we have also included the files with the modifications covered in this lab in the [DBClientApp/manifests](./DBClientApp/manifests) directory
+
+### Database Consumer vs Database Provider
+
+In this guide you will see throughout the use of a 'database consumer' and a 'database provider'. The former refers to the workload(s) that consume the database content, and the latter refers the workload(s) that provide the database content to the consumer.
+
+## Before you start
+
+> **NOTE**: If you have configured your <span style="color:#1E90FF;">**/etc/hosts**</span> file with the instance(s) names and IP address(es) you can use the lab name, i.e. lab17-persistent-storage in place of the node IP address. 
+
+You can open the Replicated Admin Console for your node by navigating to https://lab17-persistent-storage:8800 in a browser. Otherwise use `https://<Instance IP Address>:8800`. The password to your instance will be provided as part of the lab, or you can reset by SSHing into the node and running the below `kubectl kots` command. 
+
+
+```bash
+export FIRST_NAME=... # your first name
+ssh ${FIRST_NAME}@<Instance IP Address>
+kubectl kots reset-password -n default
+```
+
+## Inspecting the example application
+
+On the Application Dashboard of the Replicated Admin Console, click on the **Open Lab 17** button to see the UI of our sample application. You should see the following in your browser window:
+
+<img src="./content/lab17-app-ui.png" width=450></img>
+
+As we can see, the application is not able to connect to a postgres service. Our application at this time only contains the 'database consumer' portion. In the next steps, we will correct this by adding the 'database provider'. 
+
+## Adding the Database
```

Comment:
How about we do this via some existing postgres yaml / chart? Is the value here that we explain how to create an embedded postgres in k8s. Maybe we can just use [bitnami/postgres](https://bitnami.com/stack/postgresql/helm) and skip everything how this is setup?